### PR TITLE
optimize init health check and pod monitor

### PR DESCRIPTION
**Pull request by JiaminZhu**
1. optimize init health check 
Checking if service in the pod is running using cmd docker-compose -f [file] ps -q [service]
2. optimize pod monitor
Pod monitor will send finish to mesos if only infra container is running.
3. Add plugin tracing
Add a folder composetrace to keep all the compose files modified by each plugin.
It will be easy for debugging and figure out what does each plugin do.
4. Add signal driven debug mode.
Debug mode will be enable or disable by sending SIGUSR1 to dce process using cmd kill -SIGUSR1 [dce process id]

sandbox: http://10.180.72.46:5050/#/agents/05a9364a-0182-421a-8303-1c7c6858d32d-S64/browse?path=%2Fvar%2Flib%2Fmesos%2Fslaves%2F05a9364a-0182-421a-8303-1c7c6858d32d-S64%2Fframeworks%2Fbe208cac-365d-44da-88fb-9ac025dc96b8-0000%2Fexecutors%2Fcompose-pod-compose-test-0-e7b46f70-4731-4824-9ab2-792cdb175074%2Fruns%2Fac76330a-ba19-4789-8912-83619960af31

### CODE REVIEW

**Comment by kkrishna** [comment]
elaborate on point #4 with an example.

**Comment by JiaminZhu** [comment]
By running cmd kill -SIGUSR1 [pid] to switch log level. Here is an example to enable debug mode and print logs during pod monitor.
http://10.180.72.56:5050/#/agents/05a9364a-0182-421a-8303-1c7c6858d32d-S68/browse?path=%2Fvar%2Flib%2Fmesos%2Fslaves%2F05a9364a-0182-421a-8303-1c7c6858d32d-S68%2Fframeworks%2Fbe208cac-365d-44da-88fb-9ac025dc96b8-0000%2Fexecutors%2Fcompose-paypal-nstestconfig-nodeapp-a-0-ef4ffea3-b493-41c1-82bb-bf599a85dfe4%2Fruns%2F9046857a-6120-435b-8fc4-a86cb2400b77

File: dce/monitor/monitor.go [hunk 1]
```diff
@@ -30,118 +29,100 @@ import (
...
 			pod.PodContainers = append(pod.PodContainers[:i], pod.PodContainers[i+1:]...)
 			i--
 
 		}
 	}
 
-	if len(pod.PodContainers) == 0 {
-		log.Println("Pod Monitor : Finished")
+	// Send finished to mesos IF no container running or ONLY system proxy is running in the pod
+	if len(pod.PodContainers) == 0 || len(pod.PodContainers) == 1 && pod.PodContainers[0] == systemProxyId {
```
Comment (kkrishna) [inline dce/monitor/monitor.go]: add appropriate logs.. such as all pod containers exited .. or only system container remains..

File: dce/monitor/monitor.go [hunk 2]
```diff
@@ -30,118 +29,100 @@ import (
...
+
 	var err error
 
 	for i := 0; i < len(pod.PodContainers); i++ {
 		var healthy string
 		var exitCode int
+		var running bool
+
+		// Set log to debug level when trace mode is turned on
+		if config.EnableDebugMode() {
```
Comment (kkrishna) [inline dce/monitor/monitor.go]: does log setting need to be in for loop?

File: config/config.yaml [hunk 3]
```diff
@@ -6,6 +6,8 @@ launchtask:
    retryinterval: 10000
    timeout: 500000
    skippull: true
+   disabletrace: false
```
Comment (kkrishna) [inline config/config.yaml]: would composetrace suit better?

File: dce/main.go [hunk 4]
```diff
@@ -348,11 +359,31 @@ func getServices(ctx context.Context) map[string]bool {
...
 	fmt.Println("====================Genesis Executor (Go)====================")
 
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM, syscall.SIGKILL, syscall.SIGUSR1)
+	go func() {
+		for {
+			sig := <-sig
+			fmt.Printf("Received signal %s\n", sig.String())
+			if sig == syscall.SIGUSR1 {
+				fmt.Println("switch debug mode")
```
Comment (kkrishna) [inline dce/main.go]: this println can be gotten rid of . there is logging in SwitchDebugMode( )

File: config/config.yaml (hunk 3 above)
Comment (JiaminZhu) [reply config/config.yaml]: by default, we wanna enable compose trace

File: config/config.go [hunk 5]
```diff
@@ -230,3 +198,29 @@ func EnableVerbose() bool {
...
+func GetComposeHttpTimeout() int {
+	return GetConfig().GetInt(COMPOSE_HTTP_TIMEOUT)
+}
+
+func EnableDebugMode() bool {
+	return GetConfig().GetBool(DEBUG_MODE)
+}
+
+func SwitchDebugMode() {
+	if EnableDebugMode() {
```
Comment (kkrishna) [inline config/config.go]: config check is unnecessary in this method.

**Pull request merged**